rclpy/src/rclpy/service.cpp

```diff
@@ -150,29 +116,22 @@ service_send_response(py::capsule pyservice, py::object pyresponse, py::capsule
     throw py::value_error("capsule is not an rmw_request_id_t or rmw_service_info_t");
   }
 
-  rcl_ret_t ret = rcl_send_response(&(srv->service), header, ros_response.get());
+  rcl_ret_t ret = rcl_send_response(rcl_service_.get(), header, ros_response.get());
   if (ret != RCL_RET_OK) {
     throw RCLError("failed to send response");
   }
 }
 
-py::object
-service_take_request(py::capsule pyservice, py::object pyrequest_type)
+py::tuple
+Service::service_take_request(py::object pyrequest_type)
 {
-  auto srv = static_cast<rclpy_service_t *>(
-    rclpy_handle_get_pointer_from_capsule(pyservice.ptr(), "rclpy_service_t"));
-  if (!srv) {
-    throw py::error_already_set();
-  }
-
   destroy_ros_message_signature * destroy_ros_message = nullptr;
   void * taken_request_ptr = rclpy_create_from_py(pyrequest_type.ptr(), &destroy_ros_message);
```

Comment:
Looks like this code can be deleted since `rclpy::create_from_py` is being used below